File: memory_management.cpp
#include <iostream>

void allocateMemory() {
    int* numbers = new int[5];  // Dynamically allocate array

    // Fill the array with values
    for (int i = 0; i < 5; ++i) {
        numbers[i] = i + 1;
    }

    std::cout << "Numbers before deallocation: ";
    for (int i = 0; i < 5; ++i) {
        std::cout << numbers[i] << " ";
    }
    std::cout << std::endl;

    // Manually deallocate memory
    delete[] numbers;
}

int main() {
    allocateMemory();
    return 0;
}

File: MemoryManagement.java
import java.util.ArrayList;

public class MemoryManagement {
    public static void main(String[] args) {
        ArrayList<Integer> numbers = new ArrayList<>();
        
        // Dynamically allocate and fill the list
        for (int i = 1; i <= 5; i++) {
            numbers.add(i);
        }

        System.out.println("Numbers before clearing: " + numbers);

        // Clear the list and let the garbage collector reclaim the memory
        numbers.clear();
        System.out.println("Numbers after clearing: " + numbers);

        // Suggest garbage collection (may not run immediately)
        System.gc();
    }
}

